worlds/sc2/test/test_generation.py

```diff
@@ -779,3 +780,39 @@ def test_weapon_armor_upgrades_generic_upgrade_missions_no_countermeasure_needed
 
         # No additional starting inventory item placement is needed
         self.assertEqual(len(upgrade_items), 0)
+
+
+    def test_locking_required_items(self):
+        world_options = {
+            'mission_order': options.MissionOrder.option_custom,
+            'custom_mission_order': {
+                'campaign': {
+                    'goal': True,
+                    'layout': {
+                        'type': 'column',
+                        'size': 2,
+                        'missions': [
+                            {
+                                'index': 0,
+                                'mission_pool': ['Liberation Day']

```

Comment:
You can use `mission_tables.SC2Mission.LIBERATION_DAY.mission_name` here (and for Supreme below and Infinite Cycle in test_usecases.py). It's more robust to us changing the name down the line, which may happen with race-swaps. I don't think we're going to raceswap no-builds anytime soon, but if we do this is just a minor headache for whoever implements it. Would also happen if we did some kind of mission variant, like a macro version.